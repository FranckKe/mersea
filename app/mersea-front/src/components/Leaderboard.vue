<template>
  <section class="section">
    <div class="container">
      <b-field grouped group-multiline>
        <b-select v-model="perPage">
          <option value="5">5 per page</option>
          <option value="10">10 per page</option>
          <option value="15">15 per page</option>
          <option value="20">20 per page</option>
        </b-select>

        <div class="control is-flex">
          <input class="input" v-model="filter" type="search" placeholder="Search">
        </div>
      </b-field>

      <b-table
      :data="filtered"
      :paginated="true"
      :per-page="perPage"
      :current-page.sync="currentPage"
      :pagination-simple="false"
      :default-sort-direction="defaultSortDirection"
      default-sort="name">

      <template slot-scope="props">
        <b-table-column field="name" label="Name" width="40" sortable numeric>
          {{ props.row.name }}
        </b-table-column>

        <b-table-column field="reports_count" label="Number of reports" sortable>
          {{ props.row.reports_count }}
        </b-table-column>

        <b-table-column field="last_activity" label="Last activity" sortable centered>
          {{ props.row.last_activity }}
        </b-table-column>
      </template>
    </b-table>
  </div>
</section>
</template>

<script>
export default {
  data() {
    return {
      leaderboard: [],
      filter: '',
      defaultSortDirection: 'asc',
      currentPage: 1,
      perPage: 10
    }
  },
  mounted() {
    this.load()
  },
  methods: {
    load: async function() {
      try {
        await this.$http.get(`/leaderboard`).data
      } catch (e) {
        throw e
      }
    }
  },
  computed: {
    filtered: function() {
      this.filter = this.filter.toLowerCase()
      return this.leaderboard.filter(v => {
        return (
          v.name.toLowerCase().includes(this.filter) ||
          v.reports_count.toString().includes(this.filter) ||
          v.last_activity.includes(this.filter)
        )
      })
    }
  }
}
</script>

<style>
</style>
