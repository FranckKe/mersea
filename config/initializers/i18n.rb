Rails.application.config.i18n.available_locales = %i[fr en es]
Rails.application.config.i18n.default_locale = 'fr'
