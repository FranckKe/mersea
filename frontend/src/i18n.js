import Vue from 'vue'
import VueI18n from 'vue-i18n'

Vue.use(VueI18n)

function loadLocaleMessages() {
  const locales = require.context(
    './locales',
    true,
    /[A-Za-z0-9-_,\s]+\.json$/i
  )
  const messages = {}
  locales.keys().forEach(key => {
    const matched = key.match(/([a-z0-9]+)\./i)
    if (matched && matched.length > 1) {
      const locale = matched[1]
      messages[locale] = locales(key)
    }
  })
  return messages
}

// Detect browser language and strip the "region" language
// en-GB => en
let browserI18n = navigator.language
if (navigator.language.indexOf('-') >= 0)
  browserI18n = navigator.language.substring(0, navigator.language.indexOf('-'))

// Default to english if browser language is not supported
let initializedLocal = ['en', 'es', 'fr'].includes(browserI18n)
  ? browserI18n
  : 'en'

export default new VueI18n({
  locale: initializedLocal,
  fallbackLocale: 'en',
  messages: loadLocaleMessages()
})
