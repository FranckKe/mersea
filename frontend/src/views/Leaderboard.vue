<template>
  <section class="section">
    <div class="container">
      <h1 class="title is-1">{{ $t('contributions') }}</h1>

      <b-field grouped group-multiline>
      <b-select v-model="perPage">
        <option value="5">5 {{ $t('per_page') }}</option>
        <option value="10">10 {{ $t('per_page') }}</option>
        <option value="15">15 {{ $t('per_page') }}</option>
        <option value="20">20 {{ $t('per_page') }}</option>
      </b-select>

        <div class="control is-flex">
          <input class="input" v-model="filter" type="search" v-bind:placeholder="$t('search')">
        </div>
      </b-field>

      <b-table
      :data="filtered"
      :paginated="true"
      :per-page="perPage"
      :current-page.sync="currentPage"
      :pagination-simple="false"
      :default-sort-direction="defaultSortDirection"
      default-sort="name">

      <template slot-scope="props">
        <b-table-column field="name" v-bind:label="$t('name')" width="40" sortable numeric>
          {{ props.row.name }}
        </b-table-column>

        <b-table-column field="reports_count" v-bind:label="$t('number_of_reports')" sortable>
          {{ props.row.reports_count }}
        </b-table-column>

        <b-table-column field="last_activity" v-bind:label="$t('last_activity')" sortable centered>
          {{ props.row.last_activity }}
        </b-table-column>
      </template>
    </b-table>
  </div>
</section>
</template>

<script>
export default {
  data() {
    return {
      leaderboard: [],
      filter: '',
      defaultSortDirection: 'asc',
      currentPage: 1,
      perPage: 10
    }
  },
  mounted() {
    this.load()
  },
  methods: {
    load: async function() {
      try {
        let leaderboard = await this.$http.get(`/leaderboard`)
        this.leaderboard = leaderboard.data
      } catch (e) {
        throw e
      }
    }
  },
  computed: {
    filtered: function() {
      return this.leaderboard.filter(v => {
        return (
          v.name.toLowerCase().includes(this.filter.toLowerCase()) ||
          v.reports_count.toString().includes(this.filter.toLowerCase()) ||
          v.last_activity.includes(this.filter.toLowerCase())
        )
      })
    }
  }
}
</script>

<style>
</style>

<i18n>
{
  "en": {
    "contributions": "Contributions",
    "last_activity": "Last activity",
    "name": "Name",
    "number_of_reports": "Number of reports",
    "per_page": "per page",
    "search": "Search"
  },
  "fr": {
    "contributions": "Contributions",
    "last_activity": "Dernière activité",
    "name": "Nom",
    "number_of_reports": "Nombre de témoignages",
    "per_page": "par page",
    "search": "Rechercher"
  },
  "es": {
    "contributions": "Contribución",
    "last_activity": "Última actividad",
    "name": "Apellido",
    "number_of_reports": "número de testimonios",
    "per_page": "por página",
    "search": "Buscar"
  }
}
</i18n>
