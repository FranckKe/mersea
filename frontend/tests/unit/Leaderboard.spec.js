import VueTestUtils from '@vue/test-utils'
import Leaderboard from '@/views/Leaderboard.vue'
import Vuex from 'vuex'
import Buefy from 'buefy'
import i18n from '../../src/i18n'

VueTestUtils.config.mocks.$t = key => key
VueTestUtils.config.mocks.$tc = (key, count) => (key, count)

const localVue = VueTestUtils.createLocalVue()
localVue.use(Buefy)
localVue.use(Vuex)

describe('Leaderboard.vue', () => {
  let store

  beforeEach(() => {
    store = new Vuex.Store()
  })

  it('matches snapshot', () => {
    const wrapper = VueTestUtils.shallowMount(Leaderboard, {
      localVue,
      store,
      i18n
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
