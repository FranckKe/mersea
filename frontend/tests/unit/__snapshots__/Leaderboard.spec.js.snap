// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Leaderboard.vue matches snapshot 1`] = `
<section class="section">
  <div class="container">
    <bfield-stub grouped="true" groupmultiline="true" addons="true">
      <bselect-stub value="10">
        <option value="5">5 per_page</option>
        <option value="10">10 per_page</option>
        <option value="15">15 per_page</option>
        <option value="20">20 per_page</option>
      </bselect-stub>
      <div class="control is-flex"><input type="search" placeholder="search" class="input"></div>
    </bfield-stub>
    <btable-stub data="" columns="" isrowcheckable="function (){return!0}" checkedrows="" mobilecards="true" defaultsort="name" defaultsortdirection="asc" paginated="true" currentpage="1" perpage="10" rowclass="function (){return&quot;&quot;}"
      openeddetailed="" hasdetailedvisible="function (){return!0}" detailkey="" total="0"></btable-stub>
  </div>
</section>
`;
